dotnet/src/Microsoft.Agents.AI.DevUI/MetaApiExtensions.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Agents.AI.DevUI.Entities;
+
+namespace Microsoft.Agents.AI.DevUI;
+
+/// <summary>
+/// Provides extension methods for mapping the server metadata endpoint to an <see cref="IEndpointRouteBuilder"/>.
+/// </summary>
+internal static class MetaApiExtensions
+{
+    /// <summary>
+    /// Maps the HTTP API endpoint for retrieving server metadata.
+    /// </summary>
+    /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+    /// <returns>The <see cref="IEndpointConventionBuilder"/> for method chaining.</returns>
+    /// <remarks>
+    /// This extension method registers the following endpoint:
+    /// <list type="bullet">
+    /// <item><description>GET /meta - Retrieve server metadata including UI mode, version, capabilities, and auth requirements</description></item>
+    /// </list>
+    /// The endpoint is compatible with the Python DevUI frontend and provides essential
+    /// configuration information needed for proper frontend initialization.
+    /// </remarks>
+    public static IEndpointConventionBuilder MapMeta(this IEndpointRouteBuilder endpoints)
+    {
+        return endpoints.MapGet("/meta", GetMeta)
```

Comment:
Eventually, we should put the devui endpoints behind a common path prefix so they wont clash with other endpoints.